cmd/config/etcd/etcd.go

```diff
@@ -75,6 +76,25 @@ func New(cfg Config) (*clientv3.Client, error) {
 	return clientv3.New(cfg.Config)
 }
 
+func parseEndpoints(endpoints string) ([]string, bool, error) {
+	etcdEndpoints := strings.Split(endpoints, config.ValueSeparator)
+
+	var etcdSecure bool
+	for _, endpoint := range etcdEndpoints {
+		u, err := xnet.ParseHTTPURL(endpoint)
+		if err != nil {
+			return nil, false, err
+		}
+		if etcdSecure && u.Scheme == "http" {
+			return nil, false, fmt.Errorf("all endpoints should be https or http: %s", endpoint)
+		}
+		// If one of the endpoint is https, we will use https directly.
+		etcdSecure = etcdSecure || u.Scheme == "https"
+	}
+
```

Comment:
perhaps the port range can be validated here?

Reply:
Port range is validated by ParseHTTPURL

Reply:
```
➜  policy mc admin config set myminio kms_vault  auth_type="approle" state="on" key_name="my-minsio-key"  auth_approle_id="1c9bd05f-7fe3-79b5-c381-b0cf23522185" auth_approle_secret="205422e8-6815-6cf1-96d0-53c4f7e9bd45" endpoint="http://localhost:82000000"
kms_vault auth_type=approle state=on key_name=my-minsio-key auth_approle_id=1c9bd05f-7fe3-79b5-c381-b0cf23522185 auth_approle_secret=205422e8-6815-6cf1-96d0-53c4f7e9bd45 endpoint=http://localhost:82000000
Setting new key has been successful.
Please restart your server with `mc admin service restart myminio`.
```
This should be done for other external endpoints like kms and notification targets, openid url etc too

Reply:
This validation is already added for all notification targets, looks like vault doesn't use this yet.

Reply:
```
### ➜  mc admin config set myminio notify_kafka sasl_username="user1" topic="topic1" queue_dir="queuedir" sasl_enable="off" tls_enable="off" tls_client_auth="0" sasl_password="" tls_skip_verify="off" brokers=""localhost:9092222222222222"" queue_limit="0" state="off"
Setting new key has been successful.
Please restart your server with `mc admin service restart myminio`.
```
port validation is not done on commit aaa14fac3